### PR TITLE
Modifications to Image Loader

### PR DESCRIPTION
**Pull request by aaronrockmenezes**
## Description

Image Loader can now be used to load images as labels. 
[Issues: Sorting using sorted() is causing problems due to lexographic sorting, this is not what a normal OS does, leading to mismatch between labels and data].

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [x] Documentations (modification for documents)

## Checklist

- [x] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [x] Run `mypy -p deepchem` and check no errors
  - [x] Run `flake8 <modified file> --count` and check no errors
  - [x] Run `python -m doctest <modified file>` and check no errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: deepchem/data/data_loader.py [hunk 1]
```diff
@@ -1410,27 +1412,88 @@ def create_dataset(self,
         # Sort image files
         image_files = sorted(image_files)
```
Comment (aaronrockmenezes) [inline deepchem/data/data_loader.py:1413]: Doubt about the functionality of this sorted function. Similar issue below.

File: deepchem/data/data_loader.py [hunk 2]
```diff
@@ -1410,27 +1412,88 @@ def create_dataset(self,
...
+                        if extension in [".png", ".tif"]:
+                            label_image_files.append(zip_file)
+                elif extension in [".png", ".tif"]:
+                    label_image_files.append(label_file)
+                else:
+                    raise ValueError("Unsupported file format")
+            label_files = remainder
+        
+        # Sort label image files
+        label_image_files = sorted(label_image_files)
```
Comment (aaronrockmenezes) [inline deepchem/data/data_loader.py:1454]: Sorting issue

Comment (aaronrockmenezes) [reply deepchem/data/data_loader.py:1454]: ['C:\\Users\\aaron\\AppData\\Local\\Temp\\tmppny71vj7\\synthetic_000_images/10GRAY.tif', 'C:\\Users\\aaron\\AppData\\Local\\Temp\\tmppny71vj7\\synthetic_000_images/11GRAY.tif', 'C:\\Users\\aaron\\AppData\\Local\\Temp\\tmppny71vj7\\synthetic_000_images/12GRAY.tif', 'C:\\Users\\aaron\\AppData\\Local\\Temp\\tmppny71vj7\\synthetic_000_images/13GRAY.tif', 'C:\\Users\\aaron\\AppData\\Local\\Temp\\tmppny71vj7\\synthetic_000_images/14GRAY.tif', 'C:\\Users\\aaron\\AppData\\Local\\Temp\\tmppny71vj7\\synthetic_000_images/15GRAY.tif', 'C:\\Users\\aaron\\AppData\\Local\\Temp\\tmppny71vj7\\synthetic_000_images/16GRAY.tif', 'C:\\Users\\aaron\\AppData\\Local\\Temp\\tmppny71vj7\\synthetic_000_images/17GRAY.tif', 'C:\\Users\\aaron\\AppData\\Local\\Temp\\tmppny71vj7\\synthetic_000_images/18GRAY.tif', 'C:\\Users\\aaron\\AppData\\Local\\Temp\\tmppny71vj7\\synthetic_000_images/19GRAY.tif', 'C:\\Users\\aaron\\AppData\\Local\\Temp\\tmppny71vj7\\synthetic_000_images/1GRAY.tif', 'C:\\Users\\aaron\\AppData\\Local\\Temp\\tmppny71vj7\\synthetic_000_images/20GRAY.tif', 'C:\\Users\\aaron\\AppData\\Local\\Temp\\tmppny71vj7\\synthetic_000_images/2GRAY.tif', 'C:\\Users\\aaron\\AppData\\Local\\Temp\\tmppny71vj7\\synthetic_000_images/3GRAY.tif', 'C:\\Users\\aaron\\AppData\\Local\\Temp\\tmppny71vj7\\synthetic_000_images/4GRAY.tif', 'C:\\Users\\aaron\\AppData\\Local\\Temp\\tmppny71vj7\\synthetic_000_images/5GRAY.tif', 'C:\\Users\\aaron\\AppData\\Local\\Temp\\tmppny71vj7\\synthetic_000_images/6GRAY.tif', 'C:\\Users\\aaron\\AppData\\Local\\Temp\\tmppny71vj7\\synthetic_000_images/7GRAY.tif', 'C:\\Users\\aaron\\AppData\\Local\\Temp\\tmppny71vj7\\synthetic_000_images/8GRAY.tif', 'C:\\Users\\aaron\\AppData\\Local\\Temp\\tmppny71vj7\\synthetic_000_images/9GRAY.tif']

**Comment by aaronrockmenezes** [comment]
@rbharath Please Review

**Comment by shreyasvinaya** [comment]
@aaronrockmenezes please use Yapf version 0.32.0 for python linting

**Comment by shreyasvinaya** [comment]
This seems to be causing 
```bash
FAILED deepchem/data/tests/test_image_loader.py::TestImageLoader::test_multitype_zip_load - assert (2,) == (2, 768, 1024, 3)
  Right contains 3 more items, first extra item: 768
  Full diff:
  - (2, 768, 1024, 3)
  + (2,) 
  ```
  in python 3.8 unit tests

**Review by shreyasvinaya — COMMENTED** [review]
This looks good to me, the image loader test that was failing in python 3.10 is also now fixed

**Comment by aaronrockmenezes** [comment]
@rbharath Please do a final review

File: deepchem/data/data_loader.py [hunk 3]
```diff
@@ -1408,30 +1448,94 @@ def create_dataset(self,
...
+            for label_file in label_files:
+                filename, extension = os.path.splitext(label_file)
+                extension = extension.lower()
+                if os.path.isdir(label_file):
+                    dirfiles = [
+                        os.path.join(label_file, subfile)
+                        for subfile in os.listdir(label_file)
+                    ]
+                    remainder += dirfiles
+                elif extension == ".zip":
```
Comment (rbharath) [inline deepchem/data/data_loader.py:1472]: @aaronrockmenezes Do zip files have a set ordering upon extraction? The example above uses a label zip and a data directory zip. Do we know ordering is preserved or is it implementation dependent and can change?

**Review by rbharath — COMMENTED** [review]
@aaronrockmenezes This looks mostly good but have a question about zip file handling below

**Comment by aaronrockmenezes** [comment]
@rbharath Yes, the zip file will be extracted in the same order as they appear in the directory that is to be extracted.

File: deepchem/data/data_loader.py (hunk 3 above)
Comment (aaronrockmenezes) [reply deepchem/data/data_loader.py:1472]: All files within a zipped folder are extracted in the order they appear in the directory that is to be extracted. So yes, as far as I know, the order is maintained.

**Comment by rbharath** [comment]
@aaronrockmenezes Can you add a unit test that verifies this? Have sample directory, zip then use loader and verify it works correctly. Once that's added we can merge in

**Comment by aaronrockmenezes** [comment]
@rbharath Added test to verify that the order of contents is maintained. Please review

File: deepchem/data/tests/test_image_loader.py [hunk 4]
```diff
@@ -71,12 +88,34 @@ def test_png_simple_load_with_labels(self):
...
+        loader = dc.data.ImageLoader()
+        dataset = loader.create_dataset((self.face_path, self.face_path))
+        # These are the known dimensions of face.png
+        assert dataset.X.shape == (1, 768, 1024, 3)
+        assert dataset.y.shape == (1, 768, 1024, 3)
+
     def test_tif_simple_load(self):
         loader = dc.data.ImageLoader()
         dataset = loader.create_dataset(self.tif_image_path)
         # TODO(rbharath): Where are the color channels?
```
Comment (shreyasvinaya) [inline deepchem/data/tests/test_image_loader.py:101]: @rbharath do we remove this comment?

Comment (rbharath) [reply deepchem/data/tests/test_image_loader.py:101]: Yes, we should but fine to do in follow up PR

**Review by rbharath — APPROVED** [review]
LGTM

**Pull request merged**